java.lang.NoSuchMethodError: org.jboss.resteasy.reactive.server.core.parameters.converters.LocalDateParamConverter

### Describe the bug

Application failed to start when upgrading Quarkus from 2.5.1 to 2.7.1.

Default constructor is removed since 2.6:
* https://github.com/quarkusio/quarkus/blob/2.5/independent-projects/resteasy-reactive/server/runtime/src/main/java/org/jboss/resteasy/reactive/server/core/parameters/converters/LocalDateParamConverter.java
* https://github.com/quarkusio/quarkus/blob/2.6/independent-projects/resteasy-reactive/server/runtime/src/main/java/org/jboss/resteasy/reactive/server/core/parameters/converters/LocalDateParamConverter.java


```java
import java.time.LocalDate;
import javax.ws.rs.QueryParam;

public class SummaryRequest {

  @QueryParam("tradeDate")
  private LocalDate tradeDate;

  public LocalDate getTradeDate() {
    return tradeDate;
  }

  public void setTradeDate(LocalDate tradeDate) {
    this.tradeDate = tradeDate;
  }
}
```

```java
import java.util.concurrent.CompletionStage;
import javax.validation.Valid;
import javax.ws.rs.BeanParam;
import javax.ws.rs.GET;
import javax.ws.rs.Path;

public interface TradeResource {

  @Path("/summary")
  @GET
  CompletionStage<SummaryResponse> getSummary(@Valid @BeanParam SummaryRequest request);
}
```

### Expected behavior

Application starts successfully

### Actual behavior

Application failed to start with below stacktrace:

```
Caused by: java.lang.NoSuchMethodError: org.jboss.resteasy.reactive.server.core.parameters.converters.LocalDateParamConverter: method <init>()V not found
	at x.y.z.SummaryRequest.__quarkus_init_converter__tradeDate(SummaryRequest.java)
	at io.quarkus.rest.runtime.__QuarkusInit.init(Unknown Source)
	at io.quarkus.resteasy.reactive.server.runtime.ResteasyReactiveRecorder.createDeployment(ResteasyReactiveRecorder.java:122)
	at io.quarkus.deployment.steps.ResteasyReactiveProcessor$setupDeployment1557341086.deploy_0(Unknown Source)
	at io.quarkus.deployment.steps.ResteasyReactiveProcessor$setupDeployment1557341086.deploy(Unknown Source)
	at io.quarkus.runner.ApplicationImpl.<clinit>(Unknown Source)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:490)
	at java.base/java.lang.Class.newInstance(Class.java:584)
	at io.quarkus.runtime.Quarkus.run(Quarkus.java:66)
	at io.quarkus.runtime.Quarkus.run(Quarkus.java:41)
	at io.quarkus.runtime.Quarkus.run(Quarkus.java:120)
	at io.quarkus.runner.GeneratedMain.main(Unknown Source)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at io.quarkus.runner.bootstrap.StartupActionImpl$1.run(StartupActionImpl.java:103)
	... 1 more
```

### How to Reproduce?

_No response_

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

_No response_

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

2.7.1

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

--- a/independent-projects/resteasy-reactive/server/runtime/src/main/java/org/jboss/resteasy/reactive/server/core/parameters/converters/LocalDateParamConverter.java
+++ b/independent-projects/resteasy-reactive/server/runtime/src/main/java/org/jboss/resteasy/reactive/server/core/parameters/converters/LocalDateParamConverter.java
@@ -5,6 +5,11 @@ import java.time.format.DateTimeFormatter;
 
 public class LocalDateParamConverter extends TemporalParamConverter<LocalDate> {
 
+    // this can be called by generated code
+    public LocalDateParamConverter() {
+        super(DateTimeFormatter.ISO_LOCAL_DATE);
+    }
+
     public LocalDateParamConverter(DateTimeFormatter formatter) {
         super(formatter);
     }

--- a/independent-projects/resteasy-reactive/server/runtime/src/main/java/org/jboss/resteasy/reactive/server/core/parameters/converters/LocalDateTimeParamConverter.java
+++ b/independent-projects/resteasy-reactive/server/runtime/src/main/java/org/jboss/resteasy/reactive/server/core/parameters/converters/LocalDateTimeParamConverter.java
@@ -5,6 +5,11 @@ import java.time.format.DateTimeFormatter;
 
 public class LocalDateTimeParamConverter extends TemporalParamConverter<LocalDateTime> {
 
+    // this can be called by generated code
+    public LocalDateTimeParamConverter() {
+        super(DateTimeFormatter.ISO_LOCAL_DATE_TIME);
+    }
+
     public LocalDateTimeParamConverter(DateTimeFormatter formatter) {
         super(formatter);
     }

--- a/independent-projects/resteasy-reactive/server/runtime/src/main/java/org/jboss/resteasy/reactive/server/core/parameters/converters/LocalTimeParamConverter.java
+++ b/independent-projects/resteasy-reactive/server/runtime/src/main/java/org/jboss/resteasy/reactive/server/core/parameters/converters/LocalTimeParamConverter.java
@@ -5,6 +5,11 @@ import java.time.format.DateTimeFormatter;
 
 public class LocalTimeParamConverter extends TemporalParamConverter<LocalTime> {
 
+    // this can be called by generated code
+    public LocalTimeParamConverter() {
+        super(DateTimeFormatter.ISO_LOCAL_TIME);
+    }
+
     public LocalTimeParamConverter(DateTimeFormatter formatter) {
         super(formatter);
     }

--- a/independent-projects/resteasy-reactive/server/runtime/src/main/java/org/jboss/resteasy/reactive/server/core/parameters/converters/OffsetDateTimeParamConverter.java
+++ b/independent-projects/resteasy-reactive/server/runtime/src/main/java/org/jboss/resteasy/reactive/server/core/parameters/converters/OffsetDateTimeParamConverter.java
@@ -5,6 +5,11 @@ import java.time.format.DateTimeFormatter;
 
 public class OffsetDateTimeParamConverter extends TemporalParamConverter<OffsetDateTime> {
 
+    // this can be called by generated code
+    public OffsetDateTimeParamConverter() {
+        super(DateTimeFormatter.ISO_OFFSET_DATE_TIME);
+    }
+
     public OffsetDateTimeParamConverter(DateTimeFormatter formatter) {
         super(formatter);
     }

--- a/independent-projects/resteasy-reactive/server/runtime/src/main/java/org/jboss/resteasy/reactive/server/core/parameters/converters/OffsetTimeParamConverter.java
+++ b/independent-projects/resteasy-reactive/server/runtime/src/main/java/org/jboss/resteasy/reactive/server/core/parameters/converters/OffsetTimeParamConverter.java
@@ -5,6 +5,11 @@ import java.time.format.DateTimeFormatter;
 
 public class OffsetTimeParamConverter extends TemporalParamConverter<OffsetTime> {
 
+    // this can be called by generated code
+    public OffsetTimeParamConverter() {
+        super(DateTimeFormatter.ISO_OFFSET_TIME);
+    }
+
     public OffsetTimeParamConverter(DateTimeFormatter formatter) {
         super(formatter);
     }

--- a/independent-projects/resteasy-reactive/server/runtime/src/main/java/org/jboss/resteasy/reactive/server/core/parameters/converters/ZonedDateTimeParamConverter.java
+++ b/independent-projects/resteasy-reactive/server/runtime/src/main/java/org/jboss/resteasy/reactive/server/core/parameters/converters/ZonedDateTimeParamConverter.java
@@ -5,6 +5,11 @@ import java.time.format.DateTimeFormatter;
 
 public class ZonedDateTimeParamConverter extends TemporalParamConverter<ZonedDateTime> {
 
+    // this can be called by generated code
+    public ZonedDateTimeParamConverter() {
+        super(DateTimeFormatter.ISO_ZONED_DATE_TIME);
+    }
+
     public ZonedDateTimeParamConverter(DateTimeFormatter formatter) {
         super(formatter);
     }

--- a/independent-projects/resteasy-reactive/server/runtime/src/main/java/org/jboss/resteasy/reactive/server/handlers/ParameterHandler.java
+++ b/independent-projects/resteasy-reactive/server/runtime/src/main/java/org/jboss/resteasy/reactive/server/handlers/ParameterHandler.java
@@ -55,6 +55,7 @@ public class ParameterHandler implements ServerRestHandler {
                 handleResult(result, requestContext, false);
             }
         } catch (Exception e) {
+            log.debug("Error occurred during parameter extraction", e);
             if (e instanceof WebApplicationException) {
                 throw e;
             } else {

--- a/independent-projects/resteasy-reactive/server/vertx/src/test/java/org/jboss/resteasy/reactive/server/vertx/test/simple/BeanParamWithLocalDateTimeTest.java
+++ b/independent-projects/resteasy-reactive/server/vertx/src/test/java/org/jboss/resteasy/reactive/server/vertx/test/simple/BeanParamWithLocalDateTimeTest.java
@@ -1,0 +1,57 @@
+package org.jboss.resteasy.reactive.server.vertx.test.simple;
+
+import io.restassured.RestAssured;
+import java.time.LocalDateTime;
+import java.util.function.Supplier;
+import javax.ws.rs.BeanParam;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.QueryParam;
+import org.hamcrest.Matchers;
+import org.jboss.resteasy.reactive.server.vertx.test.framework.ResteasyReactiveUnitTest;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+public class BeanParamWithLocalDateTimeTest {
+
+    @RegisterExtension
+    static ResteasyReactiveUnitTest test = new ResteasyReactiveUnitTest()
+            .setArchiveProducer(new Supplier<>() {
+                @Override
+                public JavaArchive get() {
+                    return ShrinkWrap.create(JavaArchive.class)
+                            .addClass(HelloResource.class);
+                }
+            });
+
+    @Test
+    public void test() {
+        RestAssured.get("/hello?date=2007-12-03T10:15:30")
+                .then().statusCode(200).body(Matchers.equalTo("hello#2007"));
+    }
+
+    @Path("hello")
+    public static class HelloResource {
+
+        @GET
+        public String helloQuery(@BeanParam Container container) {
+            return "hello#" + container.getDate().getYear();
+        }
+    }
+
+    public static class Container {
+        @QueryParam("date")
+        private LocalDateTime date;
+
+        public LocalDateTime getDate() {
+            return date;
+        }
+
+        public void setDate(LocalDateTime date) {
+            this.date = date;
+        }
+    }
+
+}

